src/protagonist/DLCS.Repository/Processing/CustomerQueueRepository.cs

```diff
@@ -54,32 +54,25 @@ public CustomerQueueRepository(DlcsContext dlcsContext, ILogger<CustomerQueueRep
 
     private async Task ChangeQueueSize(int customer, string name, int amount, CancellationToken cancellationToken)
     {
-        await using var transaction =
-            await DlcsContext.Database.BeginTransactionAsync(IsolationLevel.ReadCommitted, cancellationToken);
-
         try
         {
-            var queue = await DlcsContext.Queues.SingleOrDefaultAsync(q => q.Customer == customer && q.Name == name,
-                cancellationToken: cancellationToken);
+            var updateCount = await DlcsContext.Queues
+                .Where(q => q.Customer == customer && q.Name == name)
```

Comment:
Would SingleOrDefault(..) be more expressive here?

Reply:
It would but the `UpdateFromQueryAsync()` method is an extension method for `IQueryable<T>`, and `SingleOrDefault()` would return a `T` only.